src/components/directory-content.js

```diff
@@ -568,92 +569,113 @@ const DirectoryContent = () => {
+                <FolderOpenRoundedIcon
+                    style={{ width: '100px', height: '100px' }}
+                />
+                <h1>
+                    <FormattedMessage id={'emptyDir'} />
+                </h1>
+            </div>
+        );
+    };
+
+    const renderTableContent = () => {
+        return (
+            <>
+                <ContentToolbar
+                    selectedElements={
+                        // Check selectedUuids.size here to show toolbar options only
+                        // when multi selection checkboxes are used.
+                        selectedUuids.size > 0 ? getSelectedChildren() : []
+                    }
+                />
+
+                <VirtualizedTable
+                    style={{ flexGrow: 1 }}
+                    onRowRightClick={(e) => onContextMenu(e)}
+                    onRowClick={handleRowClick}
+                    rows={getCurrentChildrenWithNotClickableRows()}
+                    columns={[
+                        {
+                            cellRenderer: selectionRenderer,
+                            dataKey: 'selected',
+                            label: '',
+                            headerRenderer: selectionHeaderRenderer,
+                            maxWidth: 60,
+                        },
+                        {
+                            width: 100,
+                            label: intl.formatMessage({
+                                id: 'elementName',
+                            }),
+                            dataKey: 'elementName',
+                            cellRenderer: nameCellRender,
+                        },
+                        {
+                            width: 100,
+                            label: intl.formatMessage({
+                                id: 'type',
+                            }),
+                            dataKey: 'type',
+                            cellRenderer: typeCellRender,
+                        },
+                        {
+                            width: 50,
+                            label: intl.formatMessage({
+                                id: 'creator',
+                            }),
+                            dataKey: 'owner',
+                            cellRenderer: creatorCellRender,
+                        },
+                    ]}
+                    sortable={true}
+                />
+            </>
+        );
+    };
+
+    const renderContent = () => {
+        if (isMissingDataAfterDirChange) {
+            return renderLoadingContent();
+        } else {
+            if (currentChildren && selectedDirectory) {
+                console.log('SBO currentChildren', currentChildren);
+                if (currentChildren.length === 0) {
+                    return renderEmptyDirContent();
+                } else {
+                    return renderTableContent();
+                }
+            }
+            return;
```

Comment:
Why this useless return at end of block ?
I'd go for keeping the `if`s aligned, with commenting/stating why `!currentChildren` and `!selectedBranches` alternatives should simply not show.

Reply:
fixed